### PR TITLE
OPA middleware enhancement proposal

### PR DESCRIPTION
**Pull request by Joffref**
As discussed in https://github.com/gofiber/awesome-fiber/pull/6, the OPA middleware provided by contrib need to be refactored.

To avoid overlap between projects, I decided to refactor the OPA middleware in this project.

This contribution is greatly inspired by this project : https://github.com/Joffref/opa-middleware

Signed-off-by: Mathis Joffre <mariusjoffre@gmail.com>

### CODE REVIEW

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
thanks for the work 
though we miss test, the readme adjustment and a bit more information about the reason for the adjustment so that everyone who finds this pull request knows what happened here

File: opafiber/fiber.go [hunk 1]
```diff
@@ -9,66 +9,62 @@ import (
...
 	if err != nil {
 		panic(fmt.Sprint("rego policy error: %w", err))
 	}
-
 	return func(c *fiber.Ctx) error {
-		input := map[string]interface{}{
-			"method": c.Method(),
-			"path":   c.Path(),
+		input := make(map[string]interface{})
+		input, err = cfg.InputCreationMethod(c)
```
Comment (mstrYoda) [inline opafiber/fiber.go:42]: No need for `input := make(map[string]interface{})` line.

`input, err := cfg.InputCreationMethod(c)` already covers it.

File: opafiber/fiber.go [hunk 2]
```diff
@@ -9,66 +9,62 @@ import (
...
 			input["query"] = queryStringData
 		}
-
-		if len(cfg.IncludeHeaders) > 0 {
-			headers := make(map[string]string)
-			for _, header := range cfg.IncludeHeaders {
-				headers[header] = c.Get(header)
-			}
-			input["headers"] = headers
+		headers := make(map[string]string)
```
Comment (mstrYoda) [inline opafiber/fiber.go:55]: It creates a map even if the user does not include headers? I think the "if" condition should remain, WDYT @ReneWerner87 @efectn ?

**Review by mstrYoda — CHANGES_REQUESTED** [review]

**Comment by mstrYoda** [comment]
LGTM

**Review by ReneWerner87 — APPROVED** [review]

**Comment by Joffref** [comment]
Do you want me to squash my commits ? @ReneWerner87

**Comment by ReneWerner87** [comment]
> Do you want that I squash my commits ? @ReneWerner87

no not really needed

**Pull request merged**